### ISSUE TITLE
BrowserLink code is not injected into web page (ASP.NET 5 RC1)

### ISSUE THREAD

**Issue by bartekkois** [aspnet/Tooling#392]
After creating a new ASP.NET 5 RC1 project BrowserLink code isn't injected into web page. app.UseBrowserLink(); is called during startup and project.json contains "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0-rc1-final".
Enviroment: W10, VS2015, ASP.NET5 RC1. Tested with Chrome and Edge.

**Comment by Eilon** [comment]
Are you running in "Development" or "Production" / "Staging"? Note that the default project template enables BrowserLink only in certain modes.

**Comment by bartekkois** [comment]
I'm runngin in Development. I even put a breakpoint in the following line to test if this method is called during startup.
`            if (env.IsDevelopment())
            {
                app.UseBrowserLink();  <<<<<<<<<<<<<< breakpoint
                app.UseDeveloperExceptionPage();
                app.UseDatabaseErrorPage();
            }`

**Comment by Eilon** [comment]
Just to confirm, are you seeing this in a plain old default new project created with RC1? And you don't see the BrowserLink JS code rendered in a default page (e.g. Home/About/Contact)?

**Comment by bartekkois** [comment]
If I create a new project and go to the Contact page this is the rendered source code:
```
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>Contact - TestBL</title>

        
            <link rel="stylesheet" href="/lib/bootstrap/dist/css/bootstrap.css" />
            <link rel="stylesheet" href="/css/site.css" />
        
        
        
    <script type="text/javascript" src="http://me.kis.scr.kaspersky-labs.com/1B74BD89-2A22-4B93-B451-1C9E1052A0EC/main.js" charset="UTF-8"></script></head>
    <body>
        <div class="navbar navbar-inverse navbar-fixed-top">
            <div class="container">
                <div class="navbar-header">
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="/">TestBL</a>
                </div>
                <div class="navbar-collapse collapse">
                    <ul class="nav navbar-nav">
                        <li><a href="/">Home</a></li>
                        <li><a href="/Home/About">About</a></li>
                        <li><a href="/Home/Contact">Contact</a></li>
                    </ul>
                </div>
            </div>
        </div>
        <div class="container body-content">
            <h2>Contact.</h2>
<h3>Your contact page.</h3>

<address>
    One Microsoft Way<br />
    Redmond, WA 98052-6399<br />
    <abbr title="Phone">P:</abbr>
    425.555.0100
</address>

<address>
    <strong>Support:</strong> <a href="mailto:Support@example.com">Support@example.com</a><br />
    <strong>Marketing:</strong> <a href="mailto:Marketing@example.com">Marketing@example.com</a>
</address>

            <hr />
            <footer>
                <p>&copy; 2016 - TestBL</p>
            </footer>
        </div>

        
            <script src="/lib/jquery/dist/jquery.js"></script>
            <script src="/lib/bootstrap/dist/js/bootstrap.js"></script>
            <script src="/js/site.js?v=EWaMeWsJBYWmL2g_KkgXZQ5nPe-a3Ichp0LEgzXczKo"></script>
        
        

        
    </body>
</html>
```

**Comment by Eilon** [comment]
Weird - I just tried with RC1 and I got the BrowserLink markup. Here's the bottom of the rendered Contact page I get:

```html
    
        <script src="/lib/jquery/dist/jquery.js"></script>
        <script src="/lib/bootstrap/dist/js/bootstrap.js"></script>
        <script src="/js/site.js?v=EWaMeWsJBYWmL2g_KkgXZQ5nPe-a3Ichp0LEgzXczKo"></script>
    
    

    

<!-- Visual Studio Browser Link -->
<script type="application/json" id="__browserLink_initializationData">
    {"requestId":"29e0cc3419714fbfb9777e7c8d516b58","requestMappingFromServer":false}
</script>
<script type="text/javascript" src="http://localhost:1583/85008cdce5b54b3187c52ed350e2beff/browserLink" async="async"></script>
<!-- End Browser Link -->

</body>
</html>
```

**Comment by Eilon** [comment]
@jodavis any ideas on this?

**Comment by bartekkois** [comment]
Anyone can help me with that? This is a clean install on my new computer. Is thare any way I can debug that problem or reinstall something?

**Comment by Eilon** [comment]
@barrytang anyone on your team can take a look at this?

**Comment by sayedihashimi** [comment]
Assigned to @madskristensen

**Comment by barrytang** [comment]
Adding @jodavis as well.

**Comment by jodavis** [comment]
Could be related to #3 here: https://github.com/aspnet/Tooling/blob/master/known-issues.md

If that doesn't work, do you catch any first-chance exceptions in the project that have "BrowserLink" in the callstack? You may have to disable "Just My Code".

If you don't see any exceptions, you can check in the registry for HKLM\Software\WOW6432Node\Microsoft\Browser Link\VisualStudio_14.0. There should be values for "dnx451" and "dnxcore50" that contain file paths that point to installed DLLs. If any of that is missing, Browser Link will not start.

**Comment by bartekkois** [comment]
Repairing ASP.NET 5 RC1 Update 1 helped.

**Comment by dealdiane** [comment]
I still have this problem. 

I've got a project that is still running RC1. There are plans to update it to 1.0 in the future but for now we're stuck with it. When I checked the registry, I can't see any dnx* values in there. I'm assuming they were removed when I installed the most recent tooling. 

How do I fix this? Can I install an older version of the browser link module and add registry values that point to that?
